subsys/bluetooth/services/ots/ots_client.c

```diff
@@ -134,6 +139,28 @@ static void tx_done(struct bt_gatt_ots_l2cap *l2cap_ctx,
 	BT_ERR("Unexpected call, context: %p, conn: %p", l2cap_ctx, (void *)conn);
 }
 
+static void write_obj_tx_done(struct bt_gatt_ots_l2cap *l2cap_ctx,
+		    struct bt_conn *conn)
+{
+	int err;
+	size_t written;
+
+	written = cur_inst->sent_size;
+	BT_DBG("ctx: %p, conn: %p, written: %d",
+	       l2cap_ctx, (void *)conn, written);
+
+	err = bt_gatt_ots_l2cap_disconnect(l2cap_ctx);
+	if (err < 0) {
+		BT_WARN("Disconnecting L2CAP returned error %d", err);
+	}
+
+	if (cur_inst->otc_inst->cb->obj_data_written) {
```

Comment:
Should we also check if `cur_inst->otc_inst` and `cur_inst->otc_inst->cb` are non-NULL?

Reply:
Add `cur_inst->otc_inst` and `cur_inst->otc_inst->cb` NULL check here.